src/wallet/api/wallet2_api.h

```diff
@@ -936,6 +942,32 @@ struct Wallet
      */
     virtual bool scanTransactions(const std::vector<std::string> &txids) = 0;
 
+    /*!
+     * \brief setupBackgroundSync       - setup background sync mode with just a view key
+     * \param background_sync_type      - the mode the wallet background syncs in
+     * \param wallet_password
+     * \param background_cache_password - custom password to encrypt background cache, only needed for custom password background sync type
+     * \return                          - true on success
+     */
+    virtual bool setupBackgroundSync(const BackgroundSyncType background_sync_type, const std::string &wallet_password, const optional<std::string> &background_cache_password = optional<std::string>()) = 0;
```

Comment:
Mixing virtual methods with default parameters can usually lead to unexpected results.  The reason is that a virtual function is called on the dynamic type of the object, while the default parameter values are based on the
static type.

See the following as an example: https://godbolt.org/z/qf65bohP1

Can you pursue an alternative approach here (not mix the two)?

Reply:
In the latest I removed the default instantiation from the virtual method in the base class, but kept it in the derived class. The discrepancy feels a little odd, but also seems less likely to cause an issue as described in your example. I would like to keep the default param in the API for devs who don't want to concern themselves with the "custom background password" option.

Another thing to keep in mind is that this pattern (default params in virtual methods) is all over the Monero code base including in virtual methods of this Wallet class, so all devs reusing these interfaces (including this one) need to be aware of this regardless.

Reply:
Sounds good, moving it out of the base class makes sense.